l2l/tests/test_gd_optimizer.py

```diff
@@ -1,53 +1,41 @@
 from l2l.optimizers.gradientdescent.optimizer import GradientDescentOptimizer
 from l2l.optimizers.gradientdescent.optimizer import RMSPropParameters
+from .test_optimizer import OptimizerTestCase
 
-from l2l.optimizees.functions.benchmarked_functions import BenchmarkedFunctions
-from l2l.optimizees.functions.optimizee import FunctionGeneratorOptimizee
+from l2l import list_to_dict
 
 
-class GDOptimizerTestCase(unittest.TestCase):
+class GDOptimizerTestCase(OptimizerTestCase):
 
-    def setUp(self):
-        name = "test_trajectory"
-        self.env = Environment(
-            trajectory=name,
-            filename=".",
-            file_title='{} data'.format(name),
-            comment='{} data'.format(name),
-            add_time=True,
-            automatic_storing=True,
-            log_stdout=False,  # Sends stdout to logs
-        )
-        self.trajectory = self.env.trajectory
-        ## Benchmark function
-        function_id = 14
-        bench_functs = BenchmarkedFunctions()
-        (benchmark_name, benchmark_function), benchmark_parameters = \
-            bench_functs.get_function_by_index(function_id, noise=True)
+    def test_gd(self):
+        optimizer_parameters = RMSPropParameters(learning_rate=0.01, exploration_step_size=0.01,
+                                       n_random_steps=1, momentum_decay=0.5,
+                                       n_iteration=1, stop_criterion=np.Inf, seed=99)
 
-        optimizee_seed = 1
-        self.optimizee = FunctionGeneratorOptimizee(self.trajectory, benchmark_function, seed=optimizee_seed)
-
-    def test_setup(self):
-
-        parameters = RMSPropParameters(learning_rate=0.01, exploration_step_size=0.01,
-                                       n_random_steps=5, momentum_decay=0.5,
-                                       n_iteration=100, stop_criterion=np.Inf, seed=99)
-
-        optimizer = GradientDescentOptimizer(self.trajectory, optimizee_create_individual=self.optimizee.create_individual,
+        optimizer = GradientDescentOptimizer(self.trajectory,
+                                             optimizee_create_individual=self.optimizee.create_individual,
                                              optimizee_fitness_weights=(0.1,),
-                                             parameters=parameters,
+                                             parameters=optimizer_parameters,
                                              optimizee_bounding_func=self.optimizee.bounding_func)
-
         self.assertIsNotNone(optimizer.parameters)
+
         try:
-            optimizer.post_process()
+
+            self.experiment.run_experiment(optimizee=self.optimizee,
+                                  optimizee_parameters=self.optimizee_parameters,
+                                  optimizer=optimizer,
+                                  optimizer_parameters=optimizer_parameters)
         except Exception:
-            self.fail()
+            self.fail(Exception.__name__)
```

Comment:
I would prefer a more pythonic way `except Exception as e ... `. Where does `self.fail` come from?

Reply:
It comes from unittest. I will change the format :)